src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/CallInstruction.cs

```diff
@@ -324,133 +324,91 @@ internal MethodInfoCallInstruction(MethodInfo target, int argumentCount)
 
         public override int ProducedStack { get { return _target.ReturnType == typeof(void) ? 0 : 1; } }
 
-        public override object Invoke(params object[] args)
-        {
-            return InvokeWorker(args);
-        }
-        public override object Invoke()
-        {
-            return InvokeWorker();
-        }
-        public override object Invoke(object arg0)
-        {
-            return InvokeWorker(arg0);
-        }
-        public override object Invoke(object arg0, object arg1)
+        public override int Run(InterpretedFrame frame)
```

Comment:
Removed indirections through virtual `Invoke` methods here. If we want to support fast paths for N-ary calls using delegates at some point - provided Core supports this - we can emit the specialized `ActionCall` and `FuncCall` instructions, which perform direct optimized stack manipulation, instead.